### PR TITLE
[SDKS-5854] Create Redis Impression Sender and add sendUniqueKeys in TelemetryConsumerSubmitter

### PR DESCRIPTION
**Pull request by nmayorsplit**
- Create Redis Impression sender
- Implement postCounters in RedisImpressionSender
- Implement synchronizeUniqueKeys  in TelemetryConsumerSubmitter

### CODE REVIEW

File: client/src/test/java/io/split/storages/pluggable/synchronizer/TelemetryConsumerSubmitterTest.java [hunk 1]
```diff
@@ -48,6 +53,27 @@ public void testTestSynchronizeConfig() throws NoSuchFieldException, IllegalAcce
...
+        List<String> keys = new ArrayList<>();
+        keys.add("key-1");
+        keys.add("key-2");
+        List<UniqueKeys.UniqueKey> uniqueKeys = new ArrayList<>();
+        uniqueKeys.add(new UniqueKeys.UniqueKey("feature-1", keys));
+        UniqueKeys uniqueKeysToSend = new UniqueKeys(uniqueKeys);
+
+        telemetrySynchronizer.synchronizeUniqueKeys(uniqueKeysToSend);
+        Gson gson = new GsonBuilder().create();
+        List<String> uniqueKeysJson = new ArrayList<>(Arrays.asList(gson.toJson(uniqueKeysToSend)));
```
Comment (gthea) [inline client/src/test/java/io/split/storages/pluggable/synchronizer/TelemetryConsumerSubmitterTest.java:76]: nit: Creating the `Gson` instance may be avoided by using the Json string directly.
```suggestion
        List<String> uniqueKeysJson = new ArrayList<>(Collections.singletonList("{\"keys\":[{\"f\":\"feature-1\",\"ks\":[\"key-1\",\"key-2\"]}]}"));
```

**Review by gthea — APPROVED** [review]

File: client/src/main/java/io/split/storages/pluggable/synchronizer/TelemetryConsumerSubmitter.java [hunk 2]
```diff
@@ -41,7 +46,8 @@ public void synchronizeStats() {
 
     @Override
     public void synchronizeUniqueKeys(UniqueKeys uniqueKeys) {
-        //TODO implements when we are using redis
+        List<String> uniqueKeysToSend = new ArrayList<>(Arrays.asList(_json.toJson(uniqueKeys)));
```
[the remark this answers is not in the document]
Comment (nmayorsplit) [reply client/src/main/java/io/split/storages/pluggable/synchronizer/TelemetryConsumerSubmitter.java]: Thank you, I changed it.

File: client/src/test/java/io/split/storages/pluggable/synchronizer/TelemetryConsumerSubmitterTest.java [hunk 3]
```diff
@@ -48,6 +53,27 @@ public void testTestSynchronizeConfig() throws NoSuchFieldException, IllegalAcce
...
+        Mockito.verify(safeUserStorageWrapper, Mockito.times(1)).pushItems(Mockito.eq("SPLITIO.uniquekeys"), Mockito.eq(uniqueKeysJson));
```
[the remark this answers is not in the document]
Comment (nmayorsplit) [reply client/src/test/java/io/split/storages/pluggable/synchronizer/TelemetryConsumerSubmitterTest.java]: I updated.

File: client/src/main/java/io/split/client/impressions/RedisImpressionSender.java [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+    private final SafeUserStorageWrapper _safeUserStorageWrapper;
+
+    private static final Logger _logger = LoggerFactory.getLogger(RedisImpressionSender.class);
+
+    public static RedisImpressionSender create(CustomStorageWrapper customStorageWrapper){
+        return new RedisImpressionSender(customStorageWrapper);
+    }
+
+    private RedisImpressionSender(CustomStorageWrapper customStorageWrapper) {
+        this._safeUserStorageWrapper = new SafeUserStorageWrapper(checkNotNull(customStorageWrapper));;
```
[the remark this answers is not in the document]
Comment (nmayorsplit) [reply client/src/main/java/io/split/client/impressions/RedisImpressionSender.java]: uh, I didn't see it. Thank you

File: client/src/test/java/io/split/storages/pluggable/synchronizer/TelemetryConsumerSubmitterTest.java (hunk 1 above)
Comment (nmayorsplit) [reply client/src/test/java/io/split/storages/pluggable/synchronizer/TelemetryConsumerSubmitterTest.java]: I updated the test case.

**Review by gthea — APPROVED** [review]

**Review by mmelograno — APPROVED** [review]

**Pull request merged**